core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackRequestValidator.java

```diff
@@ -179,5 +191,40 @@ private PlatformResourceRequestJson getRequestForEncryptionKeys(Long credentialI
         return request;
     }
 
+    private void checkResourceRequirementsIfBlueprintIsDatalakeReady(StackRequest stackRequest, ValidationResultBuilder validationBuilder) {
+        Blueprint blueprint = blueprintService.getByNameForWorkspaceId(stackRequest.getClusterRequest()
+                .getBlueprintName(), restRequestThreadLocalService.getRequestedWorkspaceId());
+        boolean sharedServiceReadyBlueprint = Optional.ofNullable((Boolean) blueprint.getTags().getMap().get("shared_services_ready")).orElse(false);
```

Comment:
You could use IfPresent here, but I don't see the usefulness of Optional in this case?

Reply:
if I would use ifPresent, then all the underlying conditions should have placed into the ifPresent which is not as clear as it is now IMHO